lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -327,6 +328,7 @@ public void finalize() {
 	<T> T ablyHttpExecute(String path, String method, Param[] headers, Param[] params, RequestBody requestBody, ResponseHandler<T> responseHandler) throws AblyException {
 		int retryCountRemaining = Hosts.isRestFallbackSupported(this.host) ? options.httpMaxRetryCount : 0;
 		String candidateHost = this.host;
+		Collections.shuffle(options.fallbackHosts);
```

Comment:
As above, I think we should shuffle just once, probably best when `options.fallbackHosts` is configured